### ISSUE TITLE
Endpoint of service lost after about 1 day 

### ISSUE THREAD

**Issue by thanhlong9x** [kubernetes/kubernetes#106198]
Hi, I have problem with endpoint object of k8s. I use api of K8s to customize the IP list of Endpoint. However, after a day, the Endpoint list disappears without the cause. Pod is not error or recreate so do not activate the update of Endpoint's IP POD list. So I want to know if a component periodically checks the Endpoint's IP error and deleting it?

**Comment by aojea** [comment]
you have to provide more data, generally a how to reproduce is the best idea 
are these endpoints that you modify custom endpoints or autogenerated for a service with a selector?

**Comment by thanhlong9x** [comment]
@aojea , you can find replace_namespaced_endpoints funtion at https://github.com/kubernetes-client/python/blob/master/kubernetes/README.md
I will describe the job I did. I will create a service with the selector label that doesn't match any label pod existing. Endpoint will be created automatically with the same name as a service. At Pod I will assign a Label to allow my application to read and realize which endpoint information needs to be updated. Next I use the upper function to change that endpoint, the information I have customized is IP and Port I want to change, the remaining information I remain

**Comment by aojea** [comment]
There is one endpoint controller that manages the endpoints associated to service with selector, is a known issue that updates on those endpoints are not reconciled, so when you update that endpoints nothing happens.
However, that endpoint "belong" to the endpoints controller, and if it resync or restart it will reconcile the endpoint 
TL;DR you can't modify endpoints managed by the endpoint controller ( associated to services with selector) and assume those changes are going to persist

**Comment by thanhlong9x** [comment]
The fact that I didn't use the selector by letting a label not refer to any pod at all. I can update the endpoint and it exists, can be used with DNS name service. But the problem is that it does not exist long for some reason, it is possible that the Endpoint Controller reconcile as you say. I want to know what do Endpoint Controller performs pod reconciliation, does it happen periodically? My pods don't have any changes that Endpoint is also lost

**Comment by aojea** [comment]
```
--min-resync-period duration     Default: 12h0m0s
--
  | The resync period in reflectors will be random between MinResyncPeriod and 2*MinResyncPeriod.
```

https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/

**Comment by caseydavenport** [comment]
I believe you're looking for this, which explains the proper way to create manual endpoints for services without selectors: https://kubernetes.io/docs/concepts/services-networking/service/#services-without-selectors

/close
